File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/PandaExecutor.java
package eu.nyerel.panda.ijplugin.runner;

import com.intellij.execution.Executor;
import com.intellij.icons.AllIcons;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public class PandaExecutor extends Executor {

	public static final String EXECUTOR_ID = "Panda Executor";

	public PandaExecutor() {
	}

	public String getToolWindowId() {
		return this.getId();
	}

	public Icon getToolWindowIcon() {
		return AllIcons.Toolwindows.ToolWindowRun;
	}

	@NotNull
	public Icon getIcon() {
		return AllIcons.Actions.Execute;
	}

	public Icon getDisabledIcon() {
		return AllIcons.Process.DisabledRun;
	}

	public String getDescription() {
		return "Run application with Panda agent";
	}

	@NotNull
	public String getActionName() {
		return "Run with Panda";
	}

	@NotNull
	public String getId() {
		return EXECUTOR_ID;
	}

	@NotNull
	public String getStartActionText() {
		return "Run with Panda";
	}

	public String getContextActionId() {
		return this.getId() + " context-action-does-not-exist";
	}

	public String getHelpId() {
		return null;
	}

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/PandaExecutor.kt
package eu.nyerel.panda.ijplugin.runner

import com.intellij.execution.Executor
import com.intellij.icons.AllIcons
import com.intellij.openapi.util.IconLoader

import javax.swing.*

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
class PandaExecutor : Executor() {

    override fun getToolWindowId(): String {
        return this.id
    }

    override fun getToolWindowIcon(): Icon {
        return AllIcons.Toolwindows.ToolWindowRun
    }

    override fun getIcon(): Icon {
        return IconLoader.getIcon("/panda-run.png", this.javaClass)
    }

    override fun getDisabledIcon(): Icon {
        return AllIcons.Process.DisabledRun
    }

    override fun getDescription(): String {
        return "Run application with Panda agent"
    }

    override fun getActionName(): String {
        return "Run with Panda"
    }

    override fun getId(): String {
        return EXECUTOR_ID
    }

    override fun getStartActionText(): String {
        return "Run with Panda"
    }

    override fun getContextActionId(): String {
        return this.id + " context-action-does-not-exist"
    }

    override fun getHelpId(): String? {
        return null
    }

    companion object {

        val EXECUTOR_ID = "Panda Executor"
    }

}
